stock_available_to_promise_release/models/stock_move.py

```diff
@@ -392,9 +392,21 @@ def _release_split(self, remaining_qty):
         # `stock.move._search_picking_for_assignation`.
         if not self.picking_id.printed:
             self.picking_id.printed = True
-        new_move = self.browse(self._split(remaining_qty))
+
+        new_move_vals = self._split(remaining_qty)
+        # In case self.product_qty <= remaining_qty, we need to use self as
+        # new move so that new picking can be assigned. If new_move_vals is
+        # empty, create method returns empty record set so _assign_picking does
+        # not assign new picking.
+        if new_move_vals:
+            new_move = self.create(new_move_vals)
+            new_move._action_confirm(merge=False)
+        else:
+            # Use self.copy() will fail the test
+            new_move = self
```

Comment:
At first I didn't understand your comment, I think it could be drop and the code rewritten simply like this:
```suggestion
        new_move = self  # Work on the current move if split doesn't occur
        new_move_vals = self._split(remaining_qty)
        if new_move_vals:
            new_move = self.create(new_move_vals)
            new_move._action_confirm(merge=False)
```